### PR TITLE
NXP i.MX95 EVK board files fix-up and improvement

### PR DESCRIPTION
**Pull request by yangbolu1991**
This pull request is to do some fix up and improvement for i.MX95 board files, like remove empty cmake files, improve doc and so on...

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

**Review by dbaluta — DISMISSED** [review]

**Review by mmahadevan108 — DISMISSED** [review]

**Comment by mmahadevan108** [comment]
@JiafeiPan , kindly review and approve

**Review by JiafeiPan — COMMENTED** [review]
Please update doc/index.rst, replace all " `  :goals: run`" to be " `  :goals: build`"

File: boards/nxp/imx95_evk/doc/index.rst [hunk 1]
```diff
@@ -92,7 +92,8 @@ The Zephyr imx95_evk_m7 board configuration supports the following hardware feat
 | UART      | on-chip    | serial port                         |
 +-----------+------------+-------------------------------------+
 
-The Zephyr imx95_evk_a55 board configuration supports the following hardware features:
+The Zephyr ``imx95_evk/mimx9596/a55`` and ``imx95_evk/mimx9596/a55/smp`` board targets supports
```
Comment (JiafeiPan) [inline boards/nxp/imx95_evk/doc/index.rst:95]: s/supports/support

**Comment by yangbolu1991** [comment]
Updated to v2. Changes,
- Added one more patch to remove redundant CONFIG_PINCTRL.
- Updated doc per @JiafeiPan suggestion.

Thanks.

**Review by dbaluta — APPROVED** [review]

**Review by JiafeiPan — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Review by LaurentiuM1234 — APPROVED** [review]

**Pull request merged**